tests/appsec/iast/test_iast.py

```diff
@@ -2,36 +2,64 @@
-        """ Test insecure hashing sha1 algorithm"""
-        r = self.weblog_get("/iast/insecure_hashing?algorithmName=sha1")
-        interfaces.library.assert_trace_exists(r)
-
-    def test_insecure_hashing_md5(self):
-        """ Test insecure hashing md5 algorithm"""
-        r = self.weblog_get("/iast/insecure_hashing?algorithmName=md5")
-        interfaces.library.assert_trace_exists(r)
-
-    def test_insecure_hashing_md4(self):
-        """ Test insecure hashing md4 algorithm"""
-        r = self.weblog_get("/iast/insecure_hashing?algorithmName=md4")
-        interfaces.library.assert_trace_exists(r)
-
-    def test_insecure_hashing_md2(self):
-        """ Test insecure hashing md2 algorithm"""
-        r = self.weblog_get("/iast/insecure_hashing?algorithmName=md2")
-        interfaces.library.assert_trace_exists(r)
+    """Verify IAST features"""
+
+    @missing_feature(library="java", reason="Need to be implement deduplicate vulnerability hashes")
+    def test_insecure_hash_remove_duplicates(self):
+        """If one line is vulnerable and it is executed multiple times (for instance in a loop) in a request, we will report only one vulnerability"""
+        r = self.weblog_get("/iast/insecure_hashing/deduplicate")
+
+        interfaces.library.expect_iast_vulnerabilities(
+            r, vulnarability_count=1, type="WEAK_HASH", location_path=self.__get_insecure_hash_source_path()
+        )
+
+    def test_insecure_hash_multiple(self):
+        """If a endpoint has multiple vulnerabilities (in diferent lines) we will report all of them"""
+        r = self.weblog_get("/iast/insecure_hashing/multiple_hash")
+
+        interfaces.library.expect_iast_vulnerabilities(
+            r, vulnarability_count=2, type="WEAK_HASH", location_path=self.__get_insecure_hash_source_path()
+        )
+
+    def test_secure_hash(self):
+        """Strong hash algorithm is not reported as insecure"""
+        r = self.weblog_get("/iast/insecure_hashing/test_algorithm?name=sha256")
+
+        interfaces.library.expect_iast_vulnerabilities(
+            r, vulnarability_count=0, type="WEAK_HASH", location_path=self.__get_insecure_hash_source_path()
+        )
+
+    def test_insecure_md5_hash(self):
+        """Test md5 weak hash algorithm reported as insecure"""
+        r = self.weblog_get("/iast/insecure_hashing/test_algorithm?name=md5")
+
+        interfaces.library.expect_iast_vulnerabilities(r, type="WEAK_HASH", evidence="md5")
+
+    @missing_feature(library="java", reason="Need to be implement sha1 hash detection")
+    def test_insecure_sha1_hash(self):
+        """Test sha1 weak hash algorithm reported as insecure"""
+        r = self.weblog_get("/iast/insecure_hashing/test_algorithm?name=sha1")
+
+        interfaces.library.expect_iast_vulnerabilities(
+            r, type="WEAK_HASH", evidence="sha1", location_path=self.__get_insecure_hash_source_path()
+        )
+
+    def __get_insecure_hash_source_path(self):
```

Comment:
Python style here. There is no private/public concept in python, but only convention. Two underscores (`__`) are reserved for class-private properties/methods. See https://docs.python.org/3/tutorial/classes.html#private-variables

The first fix would be to use only one leading underscore. But as we want to keep test class clean form any python trick, can you add at the top of the file : 

```python  

if context.library == "nodejs":
    EXPECTED_LOCATION = "/usr/app/app.js"
elif context.library == "java":
    EXPECTED_LOCATION = "com.datadoghq.system_tests.springboot.iast.utils.CryptoExamples"
else:
     EXPECTED_LOCATION = ""  # (TBD)
```